File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/entity/ChapterCover.java
package com.amakakeru.mangaworld.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;

@Data
@Entity
@Table(name = "chapter_cover")
public class ChapterCover implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "cc_id")
    private Long ccId;

    @Lob
    @Column(name = "cc_picture", nullable = false)
    private byte[] ccPicture;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "c_id", nullable = false, unique = true)
    private Chapter chapter;
}

File: Frontend/manga-world/src/Model/ChapterCover.ts
import chapter from "./Chapter";

class chapterCover {
  ccId: number;
  ccPicture: string;
  chapter: chapter;

  constructor(ccId: number, ccPicture: string, chapter: chapter) {
    this.ccId = ccId;
    this.ccPicture = ccPicture;
    this.chapter = chapter;
  }
}

export default chapterCover;